src/node_report.cc

```diff
@@ -0,0 +1,1029 @@
+                       const char* message,
+                       const char* location,
+                       char* name,
+                       MaybeLocal<Value> error) {
+  std::ostringstream oss;
+  std::string filename;
+  std::shared_ptr<PerIsolateOptions> options;
+  if (env != nullptr) options = env->isolate_data()->options();
+
+  // Obtain the current time and the pid (platform dependent)
+  TIME_TYPE tm_struct;
+#ifdef _WIN32
+  DWORD pid;
+  GetLocalTime(&tm_struct);
+#else  // UNIX, OSX
+  struct timeval time_val;
+  pid_t pid;
+  gettimeofday(&time_val, nullptr);
+  localtime_r(&time_val.tv_sec, &tm_struct);
+#endif
+  pid = uv_os_getpid();
+  // Determine the required report filename. In order of priority:
+  //   1) supplied on API 2) configured on startup 3) default generated
+  if (name != nullptr && strlen(name) > 0) {
+    // Filename was specified as API parameter, use that
+    oss << name;
+  } else if (env != nullptr && options->report_filename.length() > 0) {
+    // File name was supplied via start-up option, use that
+    oss << options->report_filename;
+  } else {
+    // Construct the report filename, with timestamp, pid and sequence number
+    oss << "node-report";
+    seq++;
+#ifdef _WIN32
+    oss << "." << std::setfill('0') << std::setw(4) << tm_struct.wYear;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wMonth;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wDay;
+    oss << "." << std::setfill('0') << std::setw(2) << tm_struct.wHour;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wMinute;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wSecond;
+    oss << "." << pid;
+    oss << "." << std::setfill('0') << std::setw(3) << seq.load();
+#else  // UNIX, OSX
+    oss << "." << std::setfill('0') << std::setw(4) << tm_struct.tm_year+1900;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_mon + 1;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_mday;
+    oss << "." << std::setfill('0') << std::setw(2) << tm_struct.tm_hour;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_min;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_sec;
+    oss << "." << pid;
+    oss << "." << std::setfill('0') << std::setw(3) << seq.load();
+#endif
+    oss << ".json";
+  }
+
+  filename = oss.str();
+  // Open the report file stream for writing. Supports stdout/err,
+  // user-specified or (default) generated name
+  std::ofstream outfile;
+  std::ostream* outstream = &std::cout;
+  if (filename == "stdout") {
+    outstream = &std::cout;
+  } else if (filename == "stderr") {
+    outstream = &std::cerr;
+  } else {
+    // Regular file. Append filename to directory path if one was specified
+    if (env != nullptr && options->report_directory.length() > 0) {
+      std::string pathname = options->report_directory;
+#ifdef _WIN32
+      pathname += "\\";
+#else
+      pathname += "/";
+#endif  //  _WIN32
```

Comment:
Nit: probably could also be extracted to a `define` or a constant.

Reply:
yes, done.